### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because:
- it fixes clipped switch in product type details view
- it fixes bug that caused querystrings to be omitted if navigation had been blocked by unsubmitted form

Note: built on top of #1913 

**PR intended to be tested with API branch:** main

### Screenshots
image (19)
Zrzut ekranu 2022-03-11 o 12 45 04

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6172 Only pathname is preserved after leaving dirty form](https://app.clickup.com/t/2549495/SALEOR-6172)

**Comment by patrys** [comment]
Task linked: [SALEOR-5995 Variant attribute switch in product type details is clipped](https://app.clickup.com/t/2549495/SALEOR-5995)

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "@material-ui/icons": "^4.11.2",
     "@material-ui/lab": "^4.0.0-alpha.58",
     "@material-ui/styles": "^4.11.4",
-    "@saleor/macaw-ui": "0.3.1",
+    "@saleor/macaw-ui": "github:saleor/macaw-ui#SALEOR-5995-switch-placement",
```
Comment (jwm0) [inline package.json:31]: Make sure to release macaw before merging this in (the branch will be deleted after it's merged and build will fail)

Comment (jwm0) [reply package.json:31]: Maybe we should introduce a CI check to make sure we don't use GH packages inside package.json

File: src/components/Form/ExitFormDialogProvider.tsx [hunk 2]
```diff
@@ -35,7 +37,9 @@ export const ExitFormDialogContext = React.createContext<ExitFormDialogData>({
   setEnableExitDialog: () => undefined,
   setExitDialogSubmitRef: () => undefined,
   shouldBlockNavigation: () => false,
-  setIsSubmitting: () => undefined
+  setIsSubmitting: () => undefined,
+  submit: () => Promise.resolve([]),
```
Comment (jwm0) [inline src/components/Form/ExitFormDialogProvider.tsx:41]: Can we just skip the entire thing and use a check in useExitFormDialogProvider to make it strictly typed?

**Review by Cloud11PL — APPROVED** [review]

File: src/orders/utils/data.ts [hunk 3]
```diff
@@ -300,6 +301,16 @@ export const isStockError = (
...
+export function addressToAddressInput<T>(
+  address: T & AddressFragment
+): AddressInput {
+  const { id, __typename, ...rest } = address;
+  return {
+    ...rest,
+    country: findInEnum(address.country.code, CountryCode)
+  };
+}
+
```
Comment (Droniu) [inline src/orders/utils/data.ts:313]: Can we delete it? It does more or less the same thing as `transformAddressToAddressInput` and it's not used anywhere

**Review by Droniu — APPROVED** [review]

File: src/components/Form/useExitFormDialog.test.tsx [hunk 4]
```diff
@@ -150,4 +150,27 @@ describe("useExitFormDialog", () => {
     expect(result.current.exit.shouldBlockNavigation()).toBe(true);
     expect(result.current.history.location.pathname).toBe(initialPath);
   });
+
+  it("navigates to full url with querystring", async () => {
+    // Given
+    const submitFn = jest.fn(() => Promise.resolve([]));
```
Comment (jwm0) [inline src/components/Form/useExitFormDialog.test.tsx:156]:
```suggestion
    const submitFn = jest.fn(async () => []);
```

**Review by jwm0 — APPROVED** [review]

File: src/components/Form/useExitFormDialog.test.tsx (hunk 4 above)
Comment (dominik-zeglen) [reply src/components/Form/useExitFormDialog.test.tsx:156]: "Potejto, potatoł" 🙃

**Pull request merged**